controllers/flashsystemcluster_controller.go

```diff
@@ -509,41 +509,35 @@ func (r *FlashSystemClusterReconciler) ensureDefaultStorageClass(instance *odfv1
 
 	err := r.Client.Get(
 		context.TODO(),
-		types.NamespacedName{Name: instance.Spec.DefaultPool.StorageClassName, Namespace: ""},
+		types.NamespacedName{Name: instance.Spec.DefaultPool.StorageClassName},
 		foundStorageClass)
 	if err != nil {
```

Comment:
you can rephrase this in such a way that you can have create call only once.

if found {
  check if it is as expected, if yes return from here, if not delete it
}

create a storage class

Reply:
yeah, good catch.

Reply:
done, thanks.